megatron/training/datasets/fim_dataset.py

```diff
@@ -0,0 +1,308 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+
+from typing import Dict, Tuple, Optional
+from dataclasses import dataclass, field
+
+import numpy as np
+import logging
+from megatron.core.datasets.gpt_dataset import GPTDataset, GPTDatasetConfig
+from megatron.core.datasets.indexed_dataset import IndexedDataset
+from megatron.core.datasets.utils import Split
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class GPTFIMDatasetConfig(GPTDatasetConfig):
+    """Configuration object for Megatron Core GPT FIM datasets"""
+
+    rate: float = None
```

Comment:
i would add `fim_` prefix to configs, so when debugging and print all configs it's more clear what it is.